### DISCUSSION TITLE
Why is there no horizontal scrollbar?

### DISCUSSION THREAD

**Discussion by scarlion1** [geany/geany#3311] [Q&A] [answered]
I have geany 1.38 installed from debian repository.  I would use it more but there is no horizontal scrollbar for some reason, so it makes some editing slow.  Any way to turn that back on?  or is it a bug I should file?  Thanks

**Comment by elextr** [comment]
The horizontal scrollbar does not show if wrapping is on since it is impossible for any line to exceed the window width in that case.

**Comment by scarlion1** [comment]
No I have wrapping turned off and there's definitely long lines here.  My mouse has one of those scroll wheels that has a left and right click and it will scroll (slowly) if I hold down the center button.

Tried turning wrapping on (Editor > Features) but it didn't actually wrap any of the lines, so something strange is going on for sure.

**Comment by elextr** [comment]
> Tried turning wrapping on (Editor > Features) but it didn't actually wrap any of the lines, so something strange is going on for sure.

If you mean in preferences, that is the default for newly opened files, it won't change already open files because the user may have set wrapping for individual files with `Document->Line Wrapping`.

**Comment by scarlion1** [comment]
Ah ok, thanks for clarifying that.  When I activated that option, the lines definitely wrap and there is no horizontal scrollbar.  However when I turn the option back off and get long lines again, still no horizontal scrollbar.

**Comment by elextr** [comment]
WFM on

```
15:00:26: Geany INFO		: Geany 1.39 (git >= 563d94d85), unknown
15:00:26: Geany INFO		: GTK 3.24.20, GLib 2.64.6
15:00:26: Geany INFO		: OS: Linux Mint 20 (ulyana)
```

please paste the same three lines from `Help->Debug Messages`

**Reply by scarlion1** [reply]
```
21:56:00.696497: Geany INFO		: Geany 1.38, en_US.UTF-8
21:56:00.696531: Geany INFO		: GTK 3.24.34, GLib 2.74.0
21:56:00.696589: Geany INFO		: OS: Debian GNU/Linux bookworm/sid
```

**Comment by elextr** [comment]
Well, unless somebody on the same platform has the same problem its hard to say why.

**Reply by scarlion1** [reply]
Weird.

I saw in that debug output is the config directory `~/.config/geany` so just tried deleting that to start with default options.  Still no horizontal scrollbar.

Also saw in `~/.config/geany` the actual config file and in there is option `show_editor_scrollbars=true` so tried changing that to `false` and it did get rid of the vertical scrollbar.  But when I turned it back to `true` just the vertical scrollbar comes back.

**Comment by elextr** [comment]
>  there is option show_editor_scrollbars=true

Yes, that is the setting `Edit->Preferences->Various->editor.show_editor_scrollbars`.

I presumed that would have no effect as your vertical scrollbar was visible.

**Comment by scarlion1** [comment, marked as answer]
Aha!  Finally found the culprit.  After noticing no horizontal scrollbar in another app, I began to think it must be something system-wide.  I haven't done _too_ much modification at that level, but I remember a while back I added some some GTK CSS modifications in `~/.config/gtk-3.0/gtk.css`.  Eventually found some settings that were matching against `paned *` and when I took that out, _voila!_ my horizontal scrollbars are back!
